internal/encryption/keyprovider/gcp_kms/config.go

```diff
@@ -0,0 +1,138 @@
+package gcp_kms
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider"
+	"github.com/opentofu/opentofu/internal/httpclient"
+	"github.com/opentofu/opentofu/version"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
+
+	kms "cloud.google.com/go/kms/apiv1"
+)
+
+type Config struct {
+	Credentials string `hcl:"credentials,optional"`
+	AccessToken string `hcl:"access_token,optional"`
+
+	ImpersonateServiceAccount          string   `hcl:"impersonate_service_account,optional"`
+	ImpersonateServiceAccountDelegates []string `hcl:"impersonate_service_account_delegates,optional"`
+
+	KMSKeyName string `hcl:"kms_encryption_key"`
+	KeySize    int    `hcl:"key_size"`
```

Comment:
The pbkdf2 provider calls it the KeyLength, so we should pick one and use it in both places.